pkg/analysis/analysis.go

```diff
@@ -16,11 +16,19 @@ var (
 )
 
 type Pass struct {
-	AnalyzerName  string
-	RootDir       string
-	SourceCodeDir string
-	ResultOf      map[*Analyzer]interface{}
-	Report        func(string, Diagnostic)
+	AnalyzerName string
```

Comment:
Since we are passing more global data to the analyzers I decided to move these parameters to its own object instead of having this function positional parameters growing ever more.

This required some minor changes in other test files and the runner but the functionality remains untouched